app-editors/helix/helix-23.10-r2.ebuild

```diff
@@ -296,8 +296,19 @@ DOCS=(
 
 PATCHES=(
 	"${FILESDIR}/helix-23.10-tree-sitter-gemini-path.patch"
+	"${FILESDIR}/helix-23.10-update-rescript-grammar.patch"
+	"${FILESDIR}/helix-23.10-update-tree-sitter-d.patch"
 )
 
+src_prepare() {
+	default
+	for grammar_subdir in runtime/grammars/sources/*; do
+		if [ -d "${grammar_subdir}" ]; then
+			git config --global --add safe.directory "${PWD}/${grammar_subdir}"
+		fi
+	done
+}
```

Comment:
o.o

I don't know, this feels weird. What does git do during building (it's not in BDEPEND anyway)? Can't you simply use **chown** instead? Is this a fatal error, and is there an upstream issue about it?

Reply:
Git is used to `git clone` the repositories of grammars specified here:

https://github.com/helix-editor/helix/blob/master/languages.toml

I can try `chown`, but from my understanding, git likes to throw that error whenever some cloned repo is owned by root. Also, not sure what I would need to chown it to, since emerge are installed by root.

Honestly, I don't like this `git clone` approach at all and I wish Helix was just downloading tarballs instead (even the ones from Github). I could file an upstream issue and PR about that.

Reply:
The other way would be just kicking out the `grammar` USE flag and tell users to fetch grammars manually with `hx --grammar fetch` after installation. That also seems cleaner TBH, because the whole mechanism of cloning random repos would happen by user's consent in user's /home, without polluting portage builddirs.

Reply:
I like that. I'm starting to feel we shouldn't micromanage it via ebuilds.

Reply:
Awesome! I will file a separate PR just for removing the `grammar` use.

Reply:
I'm also fine to remove the `grammar` USE flag.